resolwe/flow/managers/listener.py

```diff
@@ -412,18 +412,19 @@ def handle_finish(self, obj):
                                 # Copy permissions.
                                 copy_permissions(parent_data, d)
 
+                                # Copy collection.
+                                d.collection = parent_data.collection
+                                d.save()
+
+                                # Add entity to which data belongs to the collection.
                                 # Entity is added to the collection only when it is
                                 # created - when it only contains 1 Data object.
-                                entities = Entity.objects.filter(data=d).annotate(num_data=Count('data')).filter(
+                                entity = Entity.objects.filter(data=d).annotate(num_data=Count('data')).filter(
```

Comment:
If you set collection in `DataDependency.objects.create` you can add entity to collection there directly.